ExRam.Gremlinq.Providers.WebSocket/GremlinQueryEnvironmentExtensions.cs

```diff
@@ -52,14 +52,14 @@ async IAsyncEnumerator<object> Core(CancellationToken ct)
 
                     if (serializedQuery is GroovyGremlinQuery groovyScript)
                     {
-                        Log(groovyScript);
-
                         try
                         {
                             results = await _lazyGremlinClient
                                 .Value
                                 .SubmitAsync<JToken>($"{_alias}.{groovyScript.Script}", groovyScript.Bindings)
                                 .ConfigureAwait(false);
+
+                            Log(groovyScript, results.StatusAttributes);
```

Comment:
In case of an exception, nothing would have been logged.

Reply:
Ah, I may have gotten confused here as I had thought the query was being logged in the `catch` block. Would it be better to log in the finally block instead?
```csharp
var results = default(ResultSet<JToken>);
var statusAttributes = default(IReadOnlyDictionary<string, object>);

if (serializedQuery is GroovyGremlinQuery groovyScript)
{
    try
    {
        results = await _lazyGremlinClient
            .Value
            .SubmitAsync<JToken>($"{_alias}.{groovyScript.Script}", groovyScript.Bindings)
            .ConfigureAwait(false);

        statusAttributes = results.StatusAttributes;
    }
    catch (Exception ex)
    {
        _logger?.LogError(
            "Error executing Gremlin query {0}:\r\n{1}",
            groovyScript.Script,
            ex);

        throw;
    }
    finally
    {
        Log(groovyScript, statusAttributes);
    }
}
else
{
    //...
}
```
Then `Log()` could take `statusAttributes` as an optional parameter which could be checked before adding them to the logging scope.

Reply:
I'm afraid no. Must be before execution.

Reply:
Can you elaborate a little bit please? We need to submit the query in order to get the status attributes in the response, and I'm trying to log the the status attributes with the query. If we first logged the query, we would need a second log entry with the query + status attributes.

Reply:
I guess it has to be like that. You might even add an option to log the whole respone (including raw data). I'm sure people will ask for that at some point.

Reply:
Okay, I will update as requested and look at an option for the raw response as well. Thanks!